PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/messager/IntellijAzureMessage.java

```diff
@@ -0,0 +1,238 @@
+@RequiredArgsConstructor
+public class IntellijAzureMessage implements IAzureMessage {
+    static final Pattern URL_PATTERN = Pattern.compile("(https?|ftp)://(www\\d?|[a-zA-Z0-9]+)?.[a-zA-Z0-9-]+(:|.)([a-zA-Z0-9.]+|(\\d+)?)([/?:].*)?");
+    static final String DEFAULT_MESSAGE_TITLE = "Azure";
+    @Nullable
+    private String title;
+    @Nullable
+    private Object payload;
+    @Nullable
+    private Action[] actions;
+    @Nullable
+    private Boolean backgrounded;
+    @Nullable
+    @Getter
+    private Project project;
+    @Nonnull
+    @Getter
+    private final IAzureMessage original;
+
+    public IntellijAzureMessage(@Nonnull Type type, @Nonnull String message) {
+        this(new SimpleMessage(type, message));
+    }
+
+    @Nonnull
+    public String getMessage() {
+        return this.original.getMessage();
+    }
+
+    @Nonnull
+    @Override
+    public Type getType() {
+        return this.original.getType();
+    }
+
+    @Nonnull
+    @Override
+    public String getTitle() {
+        return StringUtils.firstNonBlank(this.title, this.original.getTitle(), DEFAULT_MESSAGE_TITLE);
+    }
+
+    @Nullable
+    @Override
+    public Object getPayload() {
+        return ObjectUtils.firstNonNull(this.payload, this.original.getPayload());
+    }
+
+    @Nonnull
+    @Override
+    public Action[] getActions() {
+        return ObjectUtils.firstNonNull(this.actions, this.original.getActions(), new Action[0]);
+    }
+
+    public String getContent(boolean includingDetails) {
+        if (original.getType() != IAzureMessage.Type.ERROR || !(original.getPayload() instanceof Throwable)) {
+            return IntellijAzureMessage.transformURLIntoLinks(original.getMessage());
+        }
+        final Throwable throwable = (Throwable) original.getPayload();
+        final List<IAzureOperation> operations = this.getOperations();
+        final String failure = operations.isEmpty() ? "Failed to proceed" : "Failed to " + operations.get(0).getTitle();
+        final String cause = Optional.ofNullable(IntellijAzureMessage.getCause(throwable))
+                .map(IntellijAzureMessage::transformURLIntoLinks)
+                .map(c -> String.format(", because <span style='font-style: italic;'>%s</span>", c))
+                .orElse("");
+        final String action = Optional.of(throwable)
+                .filter(t -> t instanceof AzureToolkitRuntimeException)
+                .map(t -> ((AzureToolkitRuntimeException) t).getAction())
+                .map(IntellijAzureMessage::transformURLIntoLinks)
+                .map(c -> String.format("<p>%s</p>", c))
+                .orElse("");
+        if (includingDetails) {
+            final String details = this.getDetails(operations);
+            final String style = "margin:0;margin-top:2px;padding-left:0;list-style-type:none;";
+            final String detailsMsg = StringUtils.isNotBlank(details) ? String.format("<div>Call Stack:</div><ul style='%s'>%s</ul>", style, details) : "";
```

Comment:
IMO, AzureMessage provide information like root cause and call stack, and the Messager will render it to HTML or other format